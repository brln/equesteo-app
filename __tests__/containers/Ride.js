import configureStore from 'redux-mock-store'
import React from "react"
import {fromJS, Map} from "immutable"
import renderer from "react-test-renderer"
import { Provider } from 'react-redux'
import thunk from "redux-thunk"

import RideContainer from '../../containers/Ride'


const mockStore = configureStore([thunk])

test('render', () => {
  const rideID = 'rideID1'
  const userID = 'userID1'
  const store = mockStore(fromJS({
    pouchRecords: {
      rides: {
        rideID1: {
          _id: rideID,
          userID: userID
        }
      },
      users: {
        'userID1': {
          _id: userID
        }
      },
      horseUsers: {},
      notifications: {},
      rideCarrots: {},
      rideComments: {},
      rideHorses: {},
      ridePhotos: {},
    },
    localState: {},
  }))

  const mockPouch = jest.genMockFromModule('pouchdb-react-native')
  mockPouch.get = () => { return Map() }

  const tree = renderer.create(
    <Provider store={store}>
      <RideContainer
        rideID={rideID}
      />
    </Provider>).toJSON()
  expect(tree).toMatchSnapshot()
})

