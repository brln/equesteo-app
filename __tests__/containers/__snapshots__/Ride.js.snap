// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#4b9ec1"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    style={
      Object {
        "color": "#4b9ec1",
        "textAlign": "center",
      }
    }
  >
    Loading Ride...
  </Text>
</View>
`;
